src/meta/src/controller/fragment.rs

```diff
@@ -723,6 +724,40 @@ impl CatalogController {
         )
     }
 
+    pub async fn get_job_fragment_backfill_scan_type(
+        &self,
+        job_id: ObjectId,
+    ) -> MetaResult<HashMap<model::FragmentId, PbStreamScanType>> {
+        let inner = self.inner.read().await;
+        let fragments: Vec<_> = Fragment::find()
+            .filter(fragment::Column::JobId.eq(job_id))
+            .all(&inner.db)
+            .await?;
+
+        let mut result = HashMap::new();
+
+        for fragment::Model {
+            fragment_id,
+            stream_node,
+            ..
+        } in fragments
+        {
+            let mut stream_node = stream_node.to_protobuf();
```

Comment:
I think we can use `fragment_type_mask` instead to reduce memory usage.

Reply:
We can shortcut for snapshot backfill